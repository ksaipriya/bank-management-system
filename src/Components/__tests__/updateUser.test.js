import React from 'react';
import { render } from '@testing-library/react';
import UpdateUser from '../UpdateDetails.js'
import Adapter from 'enzyme-adapter-react-16';
import ShallowRenderer from 'react-test-renderer/shallow';
import * as enzyme from 'enzyme'


enzyme.configure({ adapter: new Adapter() });
describe('Test case for update details component',()=>{
    let wrapper;
    it('header check', () => {
        wrapper = render(<UpdateUser />);
        const header = wrapper.getByTestId('title')
        expect(header.textContent).toBe('Update User')
    })

    it('proper form rendering', () => {
        const { getByText, getByLabelText } = render(<UpdateUser />)
        const name = getByText('Enter your customer id');
        expect(name).toBeInTheDocument();
        const inputUname = getByLabelText('Enter your customer id');
        expect(inputUname).toHaveAttribute('type', 'number')
    });

    it('matches snapshot', () => {
        const shallowRenderer = new ShallowRenderer();
        shallowRenderer.render(<UpdateUser/>);
        const result = shallowRenderer.getRenderOutput();
        expect(result).toMatchSnapshot();
    });
})
