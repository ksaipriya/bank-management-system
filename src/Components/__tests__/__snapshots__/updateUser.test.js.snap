// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for update details component matches snapshot 1`] = `
<Container
  fluid={false}
>
  <Form
    inline={false}
  >
    <FormGroup>
      <h4
        data-testid="title"
      >
        Update User
      </h4>
      <FormLabel
        column={false}
        id="custid"
        srOnly={false}
      >
        Enter your customer id
      </FormLabel>
      <FormControl
        aria-labelledby="custid"
        name="custid"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Please enter your customer_id to edit your details!"
        type="number"
      />
      <p />
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        variant="success"
      >
        Show Details
      </Button>
    </FormGroup>
  </Form>
</Container>
`;
